client/lib/site-spec/use-site-spec.ts

```diff
@@ -96,6 +101,6 @@ export function useSiteSpec( options: UseSiteSpecOptions = {} ) {
 			// If the loader manages a global "loaded" flag, reset it so
 			resetSiteSpecScriptState();
 		};
-		// Re-run only if the container target or handlers change.
-	}, [ container, onMessage, onError ] );
+		// Re-run only if the container target, handlers, locale, or config change.
+	}, [ container, onMessage, onError, locale, siteSpecConfig ] );
```

Comment:
Note: added the`siteSpecConfig` to fix a lint warn about its dependency. 
It does not affect the current behaviour